### PR TITLE
2.15 crash fix - card browser crash on uninitialized variable

### LINKED ISSUE #8882: 2.15.0beta4 crash on card browser open
```
java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.BaseAdapter.notifyDataSetChanged()' on a null object reference
at com.ichi2.anki.CardBrowser.onSelectionChanged(CardBrowser.java:10)
at com.ichi2.anki.CardBrowser.onCreateOptionsMenu(CardBrowser.java:34)
at android.app.Activity.onCreatePanelMenu(Activity.java:4159)
at androidx.fragment.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:1)
at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:1)
at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onCreatePanelMenu(AppCompatDelegateImpl.java:2)
at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:1)
at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:5)
at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:1)
at android.os.Handler.handleCallback(Handler.java:914)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:225)
at android.app.ActivityThread.main(ActivityThread.java:7564)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:12)
```

```

&nbsp; | --------- beginning of main
-- | --
1 | 05-19 12:31:29.364 I/AnkiDroid(10463): DeckPicker:: Sync button pressed
2 | 05-19 12:31:29.416 I/AnkiDroid(10463): Locked Collection - Collection Loading should fail
3 | 05-19 12:31:29.416 I/AnkiDroid(10463): Sync - starting sync
4 | 05-19 12:31:29.417 I/AnkiDroid(10463): ExecuteCommand: localMinutesWest
5 | 05-19 12:31:29.833 I/AnkiDroid(10463): Rust: transaction begin
6 | 05-19 12:31:29.834 I/AnkiDroid(10463): Sync: getting meta data from server
7 | 05-19 12:31:29.835 I/AnkiDroid(10463): Sync: building local meta data
8 | 05-19 12:31:29.835 I/AnkiDroid(10463): Sync: no changes - returning
9 | 05-19 12:31:29.836 I/AnkiDroid(10463): Rust: transaction rollback
10 | 05-19 12:31:29.836 I/AnkiDroid(10463): Sync - Performing media sync
11 | 05-19 12:31:30.266 I/AnkiDroid(10463): Sync Finished - Closing Collection
12 | 05-19 12:31:30.267 W/AnkiDroid(10463): DB/ Not in a transaction. Cannot end transaction.
13 | 05-19 12:31:30.267 I/AnkiDroid(10463): ExecuteCommand: closeCollection
14 | 05-19 12:31:30.271 I/AnkiDroid(10463): Closing Collection Log
15 | 05-19 12:31:30.271 I/AnkiDroid(10463): Collection closed
16 | 05-19 12:31:30.271 I/AnkiDroid(10463): Unlocked Collection
17 | 05-19 12:31:35.133 I/AnkiDroid(10463): Pull to Sync: Syncing
18 | 05-19 12:31:35.136 I/AnkiDroid(10463): Begin openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
19 | 05-19 12:31:35.137 I/AnkiDroid(10463): Opening rust backend. Server: false. Langs: 'en', path:
20 | 05-19 12:31:35.138 I/AnkiDroid(10463): Opening Collection: '/storage/emulated/0/AnkiDroid/collection.anki2' '' '' ''
21 | 05-19 12:31:35.157 I/AnkiDroid(10463): Opening Collection Log
22 | 05-19 12:31:35.172 I/AnkiDroid(10463): ExecuteCommand: localMinutesWest
23 | 05-19 12:31:35.173 I/AnkiDroid(10463): ExecuteCommand: localMinutesWest
24 | 05-19 12:31:35.174 I/AnkiDroid(10463): End openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
25 | 05-19 12:31:35.187 I/AnkiDroid(10463): Locked Collection - Collection Loading should fail
26 | 05-19 12:31:35.188 I/AnkiDroid(10463): Sync - starting sync
27 | 05-19 12:31:35.188 I/AnkiDroid(10463): ExecuteCommand: localMinutesWest
28 | 05-19 12:31:35.589 I/AnkiDroid(10463): Rust: transaction begin
29 | 05-19 12:31:35.589 I/AnkiDroid(10463): Sync: getting meta data from server
30 | 05-19 12:31:35.590 I/AnkiDroid(10463): Sync: building local meta data
31 | 05-19 12:31:35.590 I/AnkiDroid(10463): Sync: no changes - returning
32 | 05-19 12:31:35.591 I/AnkiDroid(10463): Rust: transaction rollback
33 | 05-19 12:31:35.591 I/AnkiDroid(10463): Sync - Performing media sync
34 | 05-19 12:31:36.412 I/AnkiDroid(10463): Sync Finished - Closing Collection
35 | 05-19 12:31:36.412 W/AnkiDroid(10463): DB/ Not in a transaction. Cannot end transaction.
36 | 05-19 12:31:36.412 I/AnkiDroid(10463): ExecuteCommand: closeCollection
37 | 05-19 12:31:36.414 I/AnkiDroid(10463): Closing Collection Log
38 | 05-19 12:31:36.414 I/AnkiDroid(10463): Collection closed
39 | 05-19 12:31:36.414 I/AnkiDroid(10463): Unlocked Collection
40 | 05-19 12:31:40.370 I/AnkiDroid(10463): Navigating to card browser


```

```
Hola, en la última actualización en Android, me ha dejado de hacer cosas que hacia antes, como cuándo copio una tarjeta para crear otra, no me la coloca en el mazo actual, sin embargo la última actualización anterior a esta si lo hacía. la suma de las tarjetas me lo hace mal. otra duda, este programa siempre tendré mis tarjetas?? es que estoy invietiendo mucho tiempo y no me gustaría que dentro de x años desapareciese!! estoy opositando y es la mejor opción de estudios que he encontrado. Gracias en versión de ordenador, podías arreglar copiar tarjetas, que antes si lo podía hacer. 
```

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/6fa78a3a-01c3-4447-bfbb-602921b4076f

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

CardBrowser was crashing because of an order-of-operations race between initial layout and onCollectionLoaded initializing necessary data

## Fixes
Fixes #8882

## Approach

Examined other areas that made the same last-call-before-crash and they were protecting access to the unitialized variable with a guard for collection load and null. Replicated that style in this code path and one other I saw using the possibly-unitialized-at-that-moment variable.

Also de-linted prior and de-flaked post in separate commits

## How Has This Been Tested?

It's a race condition, I can't reproduce it but I can see how it would happen.
Ran `./gradlew jacocoTestReport` against an Android S emulator locally

## Learning (optional, can help others)

Null is the billion dollar mistake.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -1875,7 +1873,7 @@ public void actualOnProgressUpdate(@NonNull CardBrowser browser, Card[] cards) {
         @Override
         protected void actualOnValidPostExecute(CardBrowser browser, BooleanGetter result) {
             browser.hideProgressBar();
-            browser.mActionBarTitle.setText(Integer.toString(browser.checkedCardCount()));
+            browser.mActionBarTitle.setText(String.format(Locale.ROOT, "%d", browser.checkedCardCount()));
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:1878]: ROOT?
It should be the users's local

Corrected in #8916

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 2]
```diff
@@ -2886,6 +2885,7 @@ void filterByTag(String... tags) {
     }
 
     @VisibleForTesting
+    @SuppressWarnings("SameParameterValue")
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:2892]: I'm sorry, but why was it useful?
If the same flag is always used, it seems to be a sign of some real trouble.
Same thing for searchQuery.
I find those warning suppression frightening.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:1878]: No, the argument is literally only a number. It is not actually a string of any kind, please let me know how formatting a string containing nothing but a number requires any non-ROOT consideration

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:2892]: I find touching code unrelated code in a bugfix frightening. But this one is just dead code that is used only in tests. Looking at the wider code may help relieve the fright?

**Pull request merged**